packages/svelte-ux/src/lib/utils/date.test.ts

```diff
@@ -0,0 +1,96 @@
+import { describe, it, expect } from 'vitest';
+import { PeriodType, formatDate, localToUtcDate, utcToLocalDate } from './date';
+
+const DATE = '2023-11-21'; // "good" default date as the day (21) is bigger than 12 (number of months). And november is a good month1 (because why not?)
+
+describe('formatDate()', () => {
+  it('should return empty string for null or undefined date', () => {
+    expect(formatDate(null)).equal('');
+    expect(formatDate(undefined)).equal('');
+  });
+
+  it('should return empty string for invalid date', () => {
+    expect(formatDate('invalid date')).equal('');
+  });
+
+  it('should format date for PeriodType.Day', () => {
+    const date = new Date(2023, 10, 21);
+    expect(formatDate(date, PeriodType.Day, 'short')).equal('11/21');
+    expect(formatDate(date, PeriodType.Day, 'long')).equal('Nov 21, 2023');
+  });
+
+  it('should format date string', () => {
+    expect(formatDate(DATE, PeriodType.Day, 'short')).equal('11/21');
+    expect(formatDate(DATE, PeriodType.Day, 'long')).equal('Nov 21, 2023');
+  });
+
+  it('should format date for PeriodType.Week', () => {
+    expect(formatDate(DATE, PeriodType.WeekSun, 'short')).equal('11/21 - 11/27');
+    expect(formatDate(DATE, PeriodType.WeekSun, 'long')).equal('11/21/2023 - 11/27/2023');
+    // QUESTION: What is WeekSun supposed to be? why is it different from WeekMon? (I would expect the first date to start on Sunday or monday? or?)
```

Comment:
Yes, this is to tied to the start of the period (week or bi-week).  Technically bi-week has 14 different starts (odd/even week, Sun - Sat).

In the past I've used this for different pay periods (some companies are Sun - Sat, some are Mon - Sun, etc) as well as other aggregations we calculated via a statistics engine.

UI wise this is controlled by...

![image](https://github.com/techniq/svelte-ux/assets/177476/f3d24148-beac-4c6e-ad42-b6394caa0d78)

Although it currently appears to be having an issue too.  I'll resolve this as I'm working on the double click preset issue.

Note, I never exposed all 14 bi-week starts in this UI as it was confusing to most users, and luckily the few customers shared the same start weeks.

Also, CalendarYear (Jan start) and FiscalYear (Oct) should really be a single year with configurable start month (Jan - Dec).  This is a bit of an implementation leak with how I used it in the past.   Fiscals typically start in October, January, or July in the states, but really they can start about anytime.

Not sure it worth refining ATM, though.

Reply:
But then: 
```ts
expect(formatDate('2023-11-21', PeriodType.WeekSun, 'short')).equal('11/21 - 11/27');
```
Should not pass no?
Shouldn't it gives `11/19 - 11/25`?

Reply:
That’s right. I’m not sure why it’s not passing TBH other than it’s typically not used. 

Usually when I’ve used these periods it’s been with an explicit start/end date and would use the Day format for each end. 

I’ve passed these period types server side to express the type of group by / aggregation to perform (ex. For the last 6 months, roll up payroll hours bi-weekly)

Reply:
I like a lot the idea of these formats. (I keep doing things on the "spot", but having a global thinking is a good thing)
I'll have a look

Reply:
Thanks.  They help a lot for both consistency, and not having to think too hard 😁.

Btw, we'll probably need to scan the repo and try to remove most/all explicit `date-fns`'s `format` within components, such as [Month](https://github.com/techniq/svelte-ux/blob/main/packages/svelte-ux/src/lib/components/Month.svelte#L94) and [DateRange](https://github.com/techniq/svelte-ux/blob/main/packages/svelte-ux/src/lib/components/DateRange.svelte#L171).  Searching for `'date-fns'` to look at imports is probably the easiest way, since the function is called `format` like Svelte UX's, and its usage is typically aliased.

LayerCake has a lot of these as well.

We can iteratively do these, just letting you know in case you start using these components and wonder why they are not in the expected language/format 😁